### PR TITLE
PR and notaro processing scripts

### PR DESCRIPTION
**Pull request by wdwatkins**
Updating the PR scripts after reprocessing the hourly data

### CODE REVIEW

File: mnt_thredds/notaro/add_soil_levels_mrso.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+#add the soil_levels variable to all mrso files
+#and associated attributes
+set -x
+for f in *mrso*.nc; 
+do
+	ncap2 -s 'soil_layer[$soil_layer]=array(0,1,$soil_layer)' $f mrso_fix/$f
+	ncatted -a axis,soil_layer,a,c,"Z" mrso_fix/$f
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/add_soil_levels_mrso.sh]: Note that you can put multiple -a commands in a ncatted call.

File: mnt_thredds/notaro/bin/compress.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    #these two lines change the missing value attribute, but they don't exist in this data	
+    #ncatted -h -O -a _FillValue,${var},m,f,${newmissing} -a missing_value,${var},m,f,${newmissing} ${SRC_DIR}/${file} ${PROC_DIR}/${file}
+    #ncatted -h -O -a _FillValue,${var},d,, -a missing_value,${var},d,, ${PROC_DIR}/${file}
+    ncap2  -O -L 0 --cnk_plc=unchunk --fl_fmt=netcdf4 -s "${var}=short(${scale}*${var}); time=long(time); mask=short(mask)" ${SRC_DIR}/${file} ${PROC_DIR}/${file}
+    ncatted -h -O -a scale_factor,${var},c,f,${scale_factor} ${PROC_DIR}/${file}
+else
+    ncap2 -h -O -L 0 --cnk_plc=unchunk --fl_fmt=netcdf4 -s "${var}=short(${var})" ${SRC_DIR}/${file} ${PROC_DIR}/${file}
+fi
+#from charlie zender on sourceforge bug post
+ncks -h -O -L 1 --cnk_map=rew ${PROC_DIR}/${file} ${DEST_DIR}/${file}
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/bin/compress.sh]: rew FTW! -- I do prefer using the expanded flags so these scripts are more legible later.

File: mnt_thredds/notaro/bin/compress_64_manual.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash 
+set -x
+#modified for 64 bit integer
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/bin/compress_64_manual.sh]: 64 bit integer?!?

File: mnt_thredds/notaro/fix_time_2.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+mkdir fixed
+for f in *{ACCESS,MRI,MIROC}*.nc;
+do
+	start=$SECONDS
+	ncks --fix_rec_dmn time $f fixed/$f
+	duration=$(( SECONDS - start ))
+	echo $f $duration seconds
+	
+done
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/fix_time_2.sh]: Note that you can symlink if you want two copies but not copied code.

File: mnt_thredds/notaro/notaro_scales_32.csv [hunk 5]
```diff
@@ -0,0 +1,37 @@
+"fileName"	"varInFile"	"min"	"max"	"absMax"	"scale"	"scale.factorAtt"
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/notaro_scales_32.csv]: Nice!

File: mnt_thredds/notaro/notaro_template_join.ncml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<netcdf xmlns="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2" enhance="false">
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/notaro_template_join.ncml]: I hate this enhance thing. Glad you got `false` in there.

File: mnt_thredds/notaro/run_compress.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+#read a file with the scales etc and run compress.sh
+scaleFile=$1
+compressScript=$2
+mkdir proc
```
Comment (dblodgett-usgs) [inline mnt_thredds/notaro/run_compress.sh]: Reading these scripts makes me miss the mac pro.

File: mnt_thredds/puerto_rico/README.txt [hunk 8]
```diff
@@ -0,0 +1,29 @@
+#data was originally in individual files by timestep in each model/domain. I separated the variables and joined across time to create files with a single variable covering the entire time range.  THE WRF data was converted to 16-bit signed integer to conserve space, with the exception of two files due to their large ranges.
```
Comment (dblodgett-usgs) [inline mnt_thredds/puerto_rico/README.txt]: Good stuff! Glad you are following through on this convention to put a readme in the main data directory that outlines this stuff.

File: mnt_thredds/puerto_rico/getScaleFactors.R [hunk 9]
```diff
@@ -0,0 +1,64 @@
+#find scale factors to retain max precision for 16 bit integer
```
Comment (dblodgett-usgs) [inline mnt_thredds/puerto_rico/getScaleFactors.R]: Didn't I see this file up above. Did it change or could you have symlinked?

File: mnt_thredds/notaro/bin/compress_64_manual.sh (hunk 3 above)
Comment (wdwatkins) [reply mnt_thredds/notaro/bin/compress_64_manual.sh]: 😄 there was one variable with a max value that didn't fit in a 32... I ended up just leaving it unpacked

**Comment by wdwatkins** [comment]
I probably just changed file names

File: mnt_thredds/puerto_rico/getScaleFactors.R (hunk 9 above)
Comment (wdwatkins) [reply mnt_thredds/puerto_rico/getScaleFactors.R]: I probably just changed file names.

**Pull request merged**